packages/connect/src/data/firmwareInfo.ts

```diff
@@ -44,220 +71,228 @@ export const getOnlineReleases = async (internalModel: DeviceModelInternal) => {
 
-    return changelog.some(item => item.required);
+    return [major_version, minor_version, patch_version];
 };
 
-const isEqual = (release: FirmwareRelease, latest: FirmwareRelease) =>
-    versionUtils.isEqual(release.version, latest.version);
+const getIntermediaryMessageRelease = (features: Features) => {
+    const { internal_model } = features;
 
-const getT1BootloaderVersion = (
-    releases2: FirmwareRelease[],
-    features: StrictFeatures,
-): VersionArray => {
-    const { bootloader_mode, major_version, minor_version, patch_version } = features;
-    const versionArray = [major_version, minor_version, patch_version] as VersionArray;
+    if (!firmwareIntermediaryReleasesConfig) {
+        throw new Error('Firmware release config not loaded.');
+    }
 
-    if (bootloader_mode) {
-        return versionArray;
+    const deviceIntermediaryReleases = firmwareIntermediaryReleasesConfig[internal_model];
+    if (!deviceIntermediaryReleases) {
+        // There are not intermediary releases for this model.
+        return undefined;
     }
+    let intermediary;
 
-    const release = releases2.find(({ version }) => versionUtils.isEqual(version, versionArray));
+    if (features.bootloader_mode) {
+        const currentBootloaderVersion = getCurrentBootloaderVersion(features);
+        intermediary = deviceIntermediaryReleases.find(inter =>
+            versionUtils.isNewer(inter.min_bootloader_version, currentBootloaderVersion),
+        );
+    } else {
+        const currentFirmwareVersion = getCurrentFirmwareVersion(features);
+        intermediary = deviceIntermediaryReleases.find(inter =>
+            versionUtils.isNewer(inter.min_firmware_version, currentFirmwareVersion),
+        );
+    }
 
-    /**
-     * FW version 1.6.0 and below don't have bootloader_version listed, so default to 1.0.0,
-     * because it doesn't matter for intermediary version calculation.
-     */
-    return release?.bootloader_version || [1, 0, 0];
+    return intermediary || undefined; // Return null if no intermediary is found
 };
 
-/**
- * v1 - bootloader < 1.8.0
- * v2 - bootloader >= 1.8.0, < 1.12.0
- * v3 - bootloader >= 1.12.0
- */
-const getIntermediaryVersion = (
-    releases2: FirmwareRelease[],
-    features: StrictFeatures,
-    offerLatest: boolean,
-): IntermediaryVersion | undefined => {
-    if (features.major_version !== 1 || offerLatest) {
-        // Intermediary is only supported on T1B1 and not needed if latest firmware is already offered
-        return;
-    }
+export const getReleaseConfig = (features: Features, firmwareType: FirmwareType) => {
+    const { internal_model } = features;
 
-    const bootloaderVersion = getT1BootloaderVersion(releases2, features);
-
-    if (versionUtils.isNewerOrEqual(bootloaderVersion, [1, 12, 0])) {
-        return 3;
+    if (internal_model === DeviceModelInternal.UNKNOWN) {
+        throw new Error('Missing Firmware release config for UNKNOWN device');
```

Comment:
What now happens if you connect an unknown device? It should somehow work - not fw update of course but connect should try to read the device and emit device event if possible. This was the purpose of unknown device. I did not check the entire call chain, maybe it is caught somewhere up

Reply:
That makes sense, so instead of throwing error what about 1c9c1840679e05f7d2080f5719940bded3cf95db ?